### PR TITLE
Deprecate unofficial Microsoft media-types and add the official ones

### LINKED ISSUE #2264: Media Type for MS office files should be updated
Maybe application/mspowerpoint is an older name.

[See MIME](https://www.iana.org/assignments/media-types/media-types.xhtml#application)

### PR DESCRIPTION
**Pull request by franktominc**
Hey guys, a first-time contributor here. Let's see how this goes

Closes #2264

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @franktominc,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by franktominc** [comment]
CLA signed

**Comment by jlprat** [comment]
OK TO TEST

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala [hunk 1]
```diff
@@ -329,7 +329,8 @@ object MediaTypes extends ObjectRegistry[(String, String), MediaType] {
 
   val `application/atom+xml`                                                      = awoc("atom+xml", "atom")
   val `application/base64`                                                        = awoc("base64", "mm", "mme")
-  val `application/excel`                                                         = abin("excel", NotCompressible, "xl", "xla", "xlb", "xlc", "xld", "xlk", "xll", "xlm", "xls", "xlt", "xlv", "xlw")
+  @deprecated("This format is unofficial and should not be used", "10.1.6")
```
Comment (jlprat) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala]: I would add in the `deprecated` message the name of the equivalent one, I guess in this case is `application/vnd.ms-excel`

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala [hunk 2]
```diff
@@ -340,7 +341,8 @@ object MediaTypes extends ObjectRegistry[(String, String), MediaType] {
   val `application/grpc+proto`                                                    = abin("grpc+proto", NotCompressible)
   val `application/lha`                                                           = abin("lha", NotCompressible, "lha")
   val `application/lzx`                                                           = abin("lzx", NotCompressible, "lzx")
-  val `application/mspowerpoint`                                                  = abin("mspowerpoint", NotCompressible, "pot", "pps", "ppt", "ppz")
+  @deprecated("This format is unofficial and should not be used", "10.1.6")
```
Comment (jlprat) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala]: Same here but with `application/vnd.ms-powerpoint`

**Comment by jlprat** [comment]
Failure was:
`akka.http.impl.engine.parsing.HttpHeaderParserLFSpec.The HttpHeaderParser (mode: LF) should not show bad performance characteristics when parameter names' hashCodes collide`
But it seems unrelated to this particular change.

PLS BUILD

**Pull request merged**